proto/rpc/v1/auth.proto

```diff
@@ -26,9 +28,12 @@ service AuthService {
 // Its sole AuthenticateTo method should be used prior to any other services that the external
 // entity's gRPC server has to offer.
 service ExternalAuthService {
-	// AuthenticateTo attempts to authenticate the caller on behalf of an entity.
-	// The resulting response contains an access token that should be used for future
-	// requests. This assumes that the caller is already authenticated to the
+	// AuthenticateTo attempts to allow the caller to authenticate to another entity.
+	// The resulting response contains an access token that with the subject
+	// as the calling entity, the audience as the other entity, and the issuer
+	// as the provider of this service. This token should be used for all
+	// future RPC requests to the other entity on the services it provides.
+	// This assumes that the caller is already authenticated to the
 	// server implementing this service.
 	rpc AuthenticateTo(AuthenticateToRequest) returns (AuthenticateToResponse) {
```

Comment:
[opt] Do you think we should/could change `AuthenticateTo` to something else so the API name represents more of what it is doing? `AssumeEntity`, `RepresentEntity`, `ImpersonateUser` etc

Reply:
That would be breaking the proto which I'm trying to avoid here. I like AssumeEntity though. If we want to change, we should file a follow up ticket since the TS SDK would have to change too.

Reply:
Might be a good task for you as well!

Reply:
True and sounds good. I'd be concerned if we don't do this before our first External API customer (not partner) uses it.

Reply:
Totally agree

Reply:
Added a Jira to not forget: https://viam.atlassian.net/browse/APP-1444